legend-engine-xt-sql-postgres-server/src/main/java/org/finos/legend/engine/postgres/Session.java

```diff
@@ -85,6 +93,28 @@ public void parse(String statementName, String query, List<Integer> paramTypes)
         parsed.put(p.name, p);
     }
 
+    /**
+     * Identify type of query and return appropriate session handler
+     * based on schema of the query.
+     *
+     * @param query SQL query to be executed
+     * @return session handler for the given query
+     */
+    private SessionHandler getSessionHandler(String query)
+    {
+        SqlBaseParser parser = SQLGrammarParser.getSqlBaseParser(query, "query");
+        List<QualifiedName> qualifiedNames = EXTRACTOR.visitSingleStatement(parser.singleStatement());
+        boolean isMetadataQuery = qualifiedNames.stream().flatMap(i -> i.parts.stream()).anyMatch(i -> (i.equalsIgnoreCase("information_schema") || i.equalsIgnoreCase("pg_catalog")));
```

Comment:
can we make string literals constants

Reply:
Also, in general it's probably safer for the constant to be on the left (i.e. ``CONSTANT.equalsIgnoreCase(value)``) to avoid an NPE risk (compared to ``value.equalsIgnoreCase(CONSTANT)``

Reply:
@shtirlets to create a tracking issue to tidy up